backend/src/main/services/LogHoursService.ts

```diff
@@ -0,0 +1,151 @@
+import PayRepository from '../repositories/PayRepository';
+import { Pay } from '../models/Pay';
+import { EmployeeHoursInputType } from '../models/EmployeeHoursInputType';
+import * as schedule from 'node-schedule';
+const log: debug.IDebugger = debug('app:userService-example');
+
+@injectable()
+export class LogHoursService {
+  private employeeHoursInputTypeRepository: EmployeeHoursInputTypeRepository;
+  private payRepository: PayRepository;
+
+  constructor(employeeHoursInputTypeRepository: EmployeeHoursInputTypeRepository, payRepository: PayRepository) {
+    log('Created new instance of LogHoursService');
+    this.employeeHoursInputTypeRepository = employeeHoursInputTypeRepository;
+    this.payRepository = payRepository;
+  }
+
+  public createLogHours = async (logHoursCreationDTO: LogHoursCreationDTO): Promise<Pay | null> => {
+    if (LogHoursService.isThereNullValueLogHoursDTO(logHoursCreationDTO)) {
+      throw new HttpException(StatusCodes.BAD_REQUEST, 'Request data is missing some values');
+    }
+
+    const inputTypeInserted = await this.employeeHoursInputTypeRepository.upsert(
+      logHoursCreationDTO.employeeHoursInputType
+    );
+
+    // delete scheduled job if exists
+    const job = schedule.scheduledJobs[logHoursCreationDTO.pay.email];
+    if (job != null) {
+      job.cancel();
+    }
+
+    if (inputTypeInserted[0].automatic) {
+      let scheduledDay = logHoursCreationDTO.employeeHoursInputType.scheduledDay;
+      if (scheduledDay === undefined) {
+        scheduledDay = ScheduledDay.SUNDAY;
+      }
+      const prevMonday = new Date();
+      const prevFriday = new Date();
+      // get last monday and last friday
+      if (scheduledDay === ScheduledDay.SATURDAY || scheduledDay === ScheduledDay.SUNDAY) {
+        prevMonday.setDate(prevMonday.getDate() - ((prevMonday.getDay() + 6) % 7));
+        prevFriday.setDate(prevFriday.getDate() - ((prevFriday.getDay() + 2) % 7));
+      }
+      // get last last monday and last friday
+      else {
+        prevMonday.setDate(prevMonday.getDate() - (((prevMonday.getDay() + 6) % 7) + 7));
+        prevFriday.setDate(prevFriday.getDate() - (prevFriday.getDay() + 2));
+      }
+      logHoursCreationDTO.pay.periodStart = prevMonday.toLocaleDateString();
+      logHoursCreationDTO.pay.periodEnd = prevFriday.toLocaleDateString();
+      const weekdayNumber = LogHoursService.getWeekdayNumber(scheduledDay);
+      // create scheduled job
+      schedule.scheduleJob(logHoursCreationDTO.pay.email, { hour: 0, minute: 0, dayOfWeek: weekdayNumber }, () => {
+        logHoursCreationDTO.pay.issueDate = new Date();
+        this.createPay(logHoursCreationDTO.pay);
+      });
+      return null;
+    } else {
+      // create pay
+      return await this.payRepository.create(logHoursCreationDTO.pay);
+    }
+  };
+
+  public createPay = async (payCreationDTO: PayCreationDTO): Promise<Pay> => {
+    return await this.payRepository.create(payCreationDTO);
```

Comment:
you don't need to await here I think because your function definition say it return a `Promise<Pay>` but if you await it actually returns a `Pay`. So it would make sense to just return the Promise and resolve it in the route layer. This comment applies to your other functions too.